server/controllers/user.controller.js

```diff
@@ -24,11 +24,19 @@ export function createUser(req, res, next) {
           if (loginErr) {
             return next(loginErr);
           }
-          res.json({
-            email: req.user.email,
-            username: req.user.username,
-            preferences: req.user.preferences,
-            id: req.user._id
+          mail.send('email-verification', {
+            body: {
+              link: `http://${req.headers.host}/api/verify?t=${auth.createVerificationToken(req.body.email)}`
```

Comment:
Since all routes that contain `/api/` only serve JSON, this link should be
```javascript
`http://${req.headers.host}/verify?t=${auth.createVerificationToken(req.body.email)}`
```
and then that page will make an AJAX request to
```
http://${req.headers.host}/api/verify?t=${auth.createVerificationToken(req.body.email)}
```